ambari-server/src/test/java/org/apache/ambari/server/api/services/mpackadvisor/MpackAdvisorHelperTest.java

```diff
@@ -0,0 +1,240 @@
+
+    MpackAdvisorCommand<MpackValidationResponse> command = mock(MpackAdvisorCommand.class);
+    MpackAdvisorRequest.MpackAdvisorRequestType requestType = MpackAdvisorRequest.MpackAdvisorRequestType.HOST_GROUPS;
+    MpackAdvisorRequest request = MpackAdvisorRequest.MpackAdvisorRequestBuilder.forStack().ofType(requestType).build();
+
+    when(command.invoke(request)).thenThrow(new MpackAdvisorException("message"));
+    doReturn(command).when(helper).createValidationCommand(request);
+    helper.validate(request);
+
+    assertTrue(false);
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testRecommend_returnsCommandResult() throws MpackAdvisorException, IOException {
+    Configuration configuration = mock(Configuration.class);
+    when(configuration.getRecommendationsArtifactsRolloverMax()).thenReturn(100);
+    MpackAdvisorRunner maRunner = mock(MpackAdvisorRunner.class);
+    AmbariMetaInfo metaInfo = mock(AmbariMetaInfo.class);
+    ServiceInfo service = mock(ServiceInfo.class);
+    when(metaInfo.getService(Mockito.anyString(), Mockito.anyString(), Mockito.anyString())).thenReturn(service);
+    when(service.getServiceAdvisorType()).thenReturn(ServiceInfo.ServiceAdvisorType.PYTHON);
+    MpackAdvisorHelper helper = mpackAdvisorHelperSpy(configuration, maRunner, metaInfo);
+
+    MpackAdvisorCommand<MpackRecommendationResponse> command = mock(MpackAdvisorCommand.class);
+    MpackRecommendationResponse expected = mock(MpackRecommendationResponse.class);
+    MpackAdvisorRequest.MpackAdvisorRequestType requestType = MpackAdvisorRequest.MpackAdvisorRequestType.HOST_GROUPS;
+
+
+    MpackAdvisorRequest request = MpackAdvisorRequest.MpackAdvisorRequestBuilder
+        .forStack()
+        .ofType(requestType)
+        .forMpackInstances(createOdsMpackInstance())
+        .build();
+
+    when(command.invoke(request)).thenReturn(expected);
+    doReturn(command).when(helper).createRecommendationCommand(ServiceInfo.ServiceAdvisorType.PYTHON, request);
+    MpackRecommendationResponse response = helper.recommend(request);
+
+    assertEquals(expected, response);
+  }
+
+  @Test(expected = MpackAdvisorException.class)
+  @SuppressWarnings("unchecked")
+  public void testRecommend_commandThrowsException_throwsException() throws MpackAdvisorException,
+      IOException {
+    Configuration configuration = mock(Configuration.class);
+    when(configuration.getRecommendationsArtifactsRolloverMax()).thenReturn(100);
+    MpackAdvisorRunner saRunner = mock(MpackAdvisorRunner.class);
+    AmbariMetaInfo metaInfo = mock(AmbariMetaInfo.class);
+    ServiceInfo service = mock(ServiceInfo.class);
+    when(metaInfo.getService(Mockito.anyString(), Mockito.anyString(), Mockito.anyString())).thenReturn(service);
+    when(service.getServiceAdvisorType()).thenReturn(ServiceInfo.ServiceAdvisorType.PYTHON);
+    MpackAdvisorHelper helper = mpackAdvisorHelperSpy(configuration, saRunner, metaInfo);
```

Comment:
There is much duplication in these test methods.  Can you please try to reduce by extracting common setup?

Reply:
Done. Added to @Before.